test/functional/wallet-encryption.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+# Copyright (c) 2016 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test Wallet encryption"""
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import (
+    assert_equal,
+    assert_raises_jsonrpc,
+)
+import os
+import shutil
+import time
+
+class WalletEncryptionTest(BitcoinTestFramework):
+
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 1
+
+    def run_test (self):
+        tmpdir = self.options.tmpdir
+        passphrase = "WalletPassphrase"
+        passphrase2 = "SecondWalletPassphrase"
+        
+        # Make sure the wallet isn't encrypted first
+        address = self.nodes[0].getnewaddress()
+        privkey = self.nodes[0].dumpprivkey(address)
+        
+        # Encrypt the wallet
+        self.nodes[0].encryptwallet(passphrase)
+        self.stop_node(0)
+        self.nodes[0] = self.start_node(0, tmpdir)
+        
+        # Test that the wallet is encrypted
+        assert_raises_jsonrpc(-13, "Please enter the wallet passphrase with walletpassphrase first", self.nodes[0].dumpprivkey, address)
+        
+        # Check that walletpassphrase works
+        self.nodes[0].walletpassphrase(passphrase, 10)
+        assert_equal(privkey, self.nodes[0].dumpprivkey(address))
+        
+        # Check that the timeout is right
+        time.sleep(10)
```

Comment:
Consider reducing the timeout in the call to walletpassphrase to shorten this test. Alternatively, you could use mocktime to avoid waiting for 10 seconds.